R/ari_stitch.R

```diff
@@ -194,68 +188,69 @@ ari_stitch <- function(images, audio,
   # ffmpeg-concat-doesnt-work-with-absolute-path
   # input_txt_path = normalizePath(input_txt_path, winslash = "\\")
 
-  ffmpeg <- ffmpeg_exec(quote = TRUE)
 
-  # set video filters
-  if (!is.null(frames_per_second)) {
-    video_filters <- c(video_filters, paste0("fps=", frames_per_second))
+  # Start organizing ffmpeg arguments here
+  ffmpeg <- ffmpeg_exec(quote = TRUE)
+  # Frames per second (fps)
+  if (!is.null(ffmpeg_args$frames_per_second)) {
+    video_filters <- c(ffmpeg_args$video_filters, paste0("fps=", ffmpeg_args$frames_per_second))
   } else {
-    video_filters <- c(video_filters, "fps=5")
+    video_filters <- c(ffmpeg_args$video_filters, "fps=5")
   }
-  if (divisible_height) {
+  # Divisible height
+  if (ffmpeg_args$divisible_height) {
     video_filters <- c(video_filters, '"scale=trunc(iw/2)*2:trunc(ih/2)*2"')
   }
 
-
   # workaround for older ffmpeg
   # https://stackoverflow.com/questions/32931685/
   # the-encoder-aac-is-experimental-but-experimental-codecs-are-not-enabled
   experimental <- FALSE
-  if (!is.null(audio_codec)) {
-    if (audio_codec == "aac") {
+  if (!is.null(ffmpeg_args$audio_codec)) {
+    if (ffmpeg_args$audio_codec == "aac") {
       experimental <- TRUE
     }
   }
-  if (deinterlace) {
+  if (ffmpeg_args$deinterlace) {
     video_filters <- c(video_filters, "yadif")
   }
   video_filters <- paste(video_filters, collapse = ",")
   video_filters <- paste0("-vf ", video_filters)
 
-  if (any(grepl("-vf", ffmpeg_opts))) {
+  if (any(grepl("-vf", ffmpeg_args$deinterlace))) {
     warning("Found video filters in ffmpeg_opts, may not be used correctly!")
   }
-  ffmpeg_opts <- c(video_filters, ffmpeg_opts)
+  ffmpeg_opts <- c(video_filters, ffmpeg_args$ffmpeg_opts)
   ffmpeg_opts <- paste(ffmpeg_opts, collapse = " ")
 
```

Comment:
Applies starting at 195: any ffmpeg argument logic should be handled by ffmpeg_args(), ideally: create a separate function that takes the result of executing ffmpeg_args() and constructs the result into a system command.